### PR TITLE
metallb pkg: Added L2Advertisement

### PR DESCRIPTION
**Pull request by evgenLevin**

### CODE REVIEW

File: pkg/metallb/l2advertisement.go [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		return nil, err
+	}
+
+	return builder.convertToStructured(unsObject)
+}
+
+// PullL2Advertisement pulls existing L2Advertisement from cluster.
+func PullL2Advertisement(apiClient *clients.Settings, name, nsname string) (*L2AdvertisementBuilder, error) {
```
Comment (kononovn) [inline pkg/metallb/l2advertisement.go:111]: Can you please move this function right after NewL2AdvertisementBuilder?

**Review by kononovn — CHANGES_REQUESTED** [review]
Just 1 minor comment

File: pkg/metallb/l2advertisement.go (hunk 1 above)
Comment (evgenLevin) [reply pkg/metallb/l2advertisement.go]: moved

**Review by kononovn — APPROVED** [review]
/lgtm

File: pkg/metallb/l2advertisement.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+}
+
+// WithNodeSelector adds the specified NodeSelectors to the L2Advertisement.
+func (builder *L2AdvertisementBuilder) WithNodeSelector(nodeSelectors []metaV1.LabelSelector) *L2AdvertisementBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating L2Advertisement %s in namespace %s with nodeSelectors: %v",
```
Comment (achuzhoy) [inline pkg/metallb/l2advertisement.go:273]: Not a matching glog entry

File: pkg/metallb/l2advertisement.go [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+}
+
+// WithIPAddressPools adds the specified IPAddressPools to the L2Advertisement.
+func (builder *L2AdvertisementBuilder) WithIPAddressPools(ipAddressPools []string) *L2AdvertisementBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating L2Advertisement %s in namespace %s with IPAddressPools: %s",
```
Comment (achuzhoy) [inline pkg/metallb/l2advertisement.go:296]: Not a matching glog entry

**Comment by achuzhoy** [comment]
3 minor comments, lgtm otherwise

File: pkg/metallb/l2advertisement.go [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+// WithIPAddressPoolsSelectors adds the specified IPAddressPoolSelectors to the L2Advertisement.
+func (builder *L2AdvertisementBuilder) WithIPAddressPoolsSelectors(
+	poolSelector []metaV1.LabelSelector) *L2AdvertisementBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating L2Advertisement %s in namespace %s with IPAddressPoolSelectors: %s",
```
[the remark this answers is not in the document]
Comment (evgenLevin) [reply pkg/metallb/l2advertisement.go]: fixed

File: pkg/metallb/l2advertisement.go (hunk 3 above)
Comment (evgenLevin) [reply pkg/metallb/l2advertisement.go]: fixed

File: pkg/metallb/l2advertisement.go (hunk 2 above)
Comment (evgenLevin) [reply pkg/metallb/l2advertisement.go]: fixed

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**